src/components/panel/panel.js

```diff
@@ -1662,56 +1685,93 @@ MdPanelPosition.prototype.absolute = function() {
   return this;
 };
 
+/**
+ * Sets the value of a position for the panel. Clears any previously set position.
+ * @param {string} position Position to set
+ * @param {string=} value Value of the position. Defaults to '0'.
+ * @returns {MdPanelPosition}
+ * @private
+ */
+MdPanelPosition.prototype._setPosition = function(position, value) {
+  if (position === 'right' || position === 'left') {
```

Comment:
I really dislike "random" strings in code, even if it won't be user visible. (Too easy to mistype when editing the code.) What do you think about making a absolutePosition enum with left, right, top, and bottom?

MdPanelPosition.absPosition.